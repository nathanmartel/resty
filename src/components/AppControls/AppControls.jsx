import React, { useState } from 'react';

const AppControls = () => {
  const [url, setUrl] = useState('');
  const [reqType, setReqType] = useState('');
  const [reqBody, setReqBody] = useState('');

  const handleSubmit = () => {
    console.log(`${reqType} | ${url} | ${reqBody} `);
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <label>URL:
          <input type="text" value={url} onChange={() => setUrl(e => e.target.value)} />
        </label>
        <label>
          <input type="radio" id="reqType-get" name="reqType" value="GET" checked={reqType === 'GET'} onChange={() => setReqType(e => e.target.value)}/>
          GET
        </label>
        <label>
          <input type="radio" id="reqType-post" name="reqType" value="POST" checked={reqType === 'POST'} onChange={() => setReqType(e => e.target.value)} />
          POST
        </label>
        <label>Body:
          <input type="textarea" value={reqBody} onChange={() => setReqBody(e => e.target.value)} />
        </label>
        <button type="submit">Submit</button>
      </form>
    </>
  );
};

export default AppControls;
