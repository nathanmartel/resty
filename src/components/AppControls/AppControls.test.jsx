import React from 'react';
import { shallow } from 'enzyme';
import AppControls from './AppControls';

describe('AppControls component', () => {
  it('renders AppControls', () => {
    const wrapper = shallow(<AppControls />);
    expect(wrapper).toMatchSnapshot();
  });
});
