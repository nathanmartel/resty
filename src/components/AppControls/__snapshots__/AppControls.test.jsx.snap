// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppControls component renders AppControls 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <label>
      URL:
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label>
      <input
        checked={false}
        id="reqType-get"
        name="reqType"
        onChange={[Function]}
        type="radio"
        value="GET"
      />
      GET
    </label>
    <label>
      <input
        checked={false}
        id="reqType-post"
        name="reqType"
        onChange={[Function]}
        type="radio"
        value="POST"
      />
      POST
    </label>
    <label>
      Body:
      <input
        onChange={[Function]}
        type="textarea"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</Fragment>
`;
